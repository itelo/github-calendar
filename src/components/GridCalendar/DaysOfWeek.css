.days-of-week-container {
  display: flex;
  flex-direction: column;
}

.day-of-week-container {
  margin-right: 4px;
  margin-top: 2px;
  height: 8px;
}

.day-of-week-font {
  font-size: 9px;
  line-height: 1;
  margin: 0;
  color: #586069;
}
