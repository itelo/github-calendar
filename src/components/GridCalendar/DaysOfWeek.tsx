import * as React from "react";
import "./DaysOfWeek.css";

export const DAYS_OF_WEEK = ["", "Mon", "", "Wed", "", "Fri", ""];

const DaysOfWeek = () => (
  <div className="days-of-week-container">
    {DAYS_OF_WEEK.map((dayOfWeek, index) => (
      <div className="day-of-week-container" key={`day-${index}`}>
        <p className="day-of-week-font" data-testid={`day-${index}`}>
          {dayOfWeek}
        </p>
      </div>
    ))}
  </div>
);

export default DaysOfWeek;
