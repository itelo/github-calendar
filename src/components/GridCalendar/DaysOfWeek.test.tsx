import React from "react";
import { render, fireEvent } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import DaysOfWeek, { DAYS_OF_WEEK } from "./DaysOfWeek";
import "jest-dom/extend-expect";

describe("<DaysOfWeek />", () => {
  test("test months passing mocked months", () => {
    const { getByTestId, container } = render(<DaysOfWeek />);

    DAYS_OF_WEEK.forEach((day, index) => {
      const content = getByTestId(`day-${index}`);
      expect(content).toHaveTextContent(day);
    });

    expect(container).toMatchSnapshot();
  });
});
