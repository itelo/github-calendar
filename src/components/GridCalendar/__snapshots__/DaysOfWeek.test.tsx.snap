// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DaysOfWeek /> test months passing mocked months 1`] = `
<div>
  <div
    class="days-of-week-container"
  >
    <div
      class="day-of-week-container"
    >
      <p
        class="day-of-week-font"
        data-testid="day-0"
      />
    </div>
    <div
      class="day-of-week-container"
    >
      <p
        class="day-of-week-font"
        data-testid="day-1"
      >
        Mon
      </p>
    </div>
    <div
      class="day-of-week-container"
    >
      <p
        class="day-of-week-font"
        data-testid="day-2"
      />
    </div>
    <div
      class="day-of-week-container"
    >
      <p
        class="day-of-week-font"
        data-testid="day-3"
      >
        Wed
      </p>
    </div>
    <div
      class="day-of-week-container"
    >
      <p
        class="day-of-week-font"
        data-testid="day-4"
      />
    </div>
    <div
      class="day-of-week-container"
    >
      <p
        class="day-of-week-font"
        data-testid="day-5"
      >
        Fri
      </p>
    </div>
    <div
      class="day-of-week-container"
    >
      <p
        class="day-of-week-font"
        data-testid="day-6"
      />
    </div>
  </div>
</div>
`;
